index.d.ts

```diff
@@ -84,3 +84,26 @@ const ab: Merge<Foo, Bar> = {a: 1, b: 2};
 ```
 */
 export type Merge<FirstType, SecondType> = Omit<FirstType, Extract<keyof FirstType, keyof SecondType>> & SecondType;
+
+/*
+Create a type that requires at least one of the given keys.
+
```

Comment:
Would you be able to mention some real-world use-cases for this type? It can be hard sometimes for users to realize where they could take advantage of a certain type.